aarch64_extension.cpp

```diff
@@ -97,6 +169,20 @@ class AArch64ArchitectureExtension : public ArchitectureHook {
     return (BNLowLevelILFlagCondition) -1;
   }
 
+  uint32_t getVectorSubregister(uint32_t baseRegister, size_t elemSize,
+                                size_t index) {
+    for (auto& regPair : mRegisters) {
+      uint32_t id = regPair.first;
+      BNRegisterInfo reg = regPair.second.second;
+
+      if (reg.fullWidthRegister == baseRegister &&
+          reg.offset == elemSize * index && reg.size == elemSize) {
+        return id;
+      }
+    }
+    return 0;
+  }
+
 public:
   explicit AArch64ArchitectureExtension(Architecture* aarch64)
```

Comment:
I think this constructor has kind of blown up now, it might be a good idea to split this off into a `bool Init()` function and then handle all errors as a `return false;`, then when Init is called in the plug-in initializer, we can print an error and not load the plug-in. You are welcome to do this, if not I can make the adjustment after the PR is merged.

Reply:
Agreed, done.